### PR TITLE
[jog_arm] Pass planning scene monitor into cpp interface

### PR DESCRIPTION
**Pull request by davetcoleman**
- Separate LOGNAME into each file
- Add unit test

I need to pass in the PSM from a larger application. Previously it required creating a new PSM even in the same process

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
> The image is still rather large: 4032x3024. Please reduce to something like 1200x.. or 1024x.. Thanks.

was this comment meant for ros-planning/moveit.ros.org#401?

File: moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_interface_base.h [hunk 1]
```diff
@@ -55,6 +55,8 @@ namespace moveit_jog_arm
 class JogInterfaceBase
 {
 public:
+  JogInterfaceBase();
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_interface_base.h]: It's somewhat weird that the constructor doesn't initialize the `planning_scene_monitor_` member.
But I see the simplicity of your approach...

File: moveit_experimental/moveit_jog_arm/src/collision_check_thread.cpp [hunk 2]
```diff
@@ -38,49 +38,38 @@
...
+  // Init collision request
   collision_detection::CollisionRequest collision_request;
   collision_request.group_name = parameters_.move_group_name;
   collision_request.distance = true;
   collision_detection::CollisionResult collision_result;
-  robot_state::RobotState& current_state = planning_scene_monitor_->getPlanningScene()->getCurrentStateNonConst();
 
-  double velocity_scale_coefficient = -log(0.001) / parameters_.collision_proximity_threshold;
+  // Copy the planning scene's version of current state into new memory
+  robot_state::RobotState current_state(getLockedPlanningSceneRO()->getCurrentState());
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/src/collision_check_thread.cpp]: Great! Better decoupling of PSM.

File: moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  void TearDown() override
+  {
+  }
+
+protected:
+  std::unique_ptr<ros::NodeHandle> nh_;
+  planning_scene_monitor::PlanningSceneMonitorPtr planning_scene_monitor_;
+};  // class TestJogCppInterface
+
+TEST_F(TestJogCppInterface, PlanningSceneTest)
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp]: These tests are not very informative yet, but they are a start :wink:

File: moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+{
+  EXPECT_TRUE(planning_scene_monitor_->getPlanningScene());
+}
+
+TEST_F(TestJogCppInterface, InitTest)
+{
+  moveit_jog_arm::JogCppApi jog_cpp_interface(planning_scene_monitor_);
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp]: What about inserting a ROS sleep for a second or so to give the threads some time to actually process data?

File: moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.test [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<launch>
+  <!-- Note, we're using fake, simulated trajectory controllers.  -->
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.test]: I don't see any controllers configured...

File: moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h [hunk 6]
```diff
@@ -49,8 +49,16 @@ namespace moveit_jog_arm
 class CollisionCheckThread
 {
 public:
+  /** \brief Constructor
+   *  \param parameters: common settings of jog_rm
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h]:
```suggestion
   *  \param parameters: common settings of jog_arm
```

File: moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_interface_base.h (hunk 1 above)
Comment (davetcoleman) [reply moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_interface_base.h]: Expanded PR description

File: moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp (hunk 3 above)
Comment (davetcoleman) [reply moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp]: I realized need to do huge refactors of jog_arm to make it testable (separate ROS and simplify threading)

File: moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp (hunk 4 above)
Comment (davetcoleman) [reply moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp]: I don't think we should add arbitrary sleeps until its proven needed.

Comment (rhaschke) [reply moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp]: Currently, the thread is immediately stopped due to immediate destruction of JogCppApi.
Adding a sleep would give the thread same time to cycle.

**Comment by davetcoleman** [comment]
@rhaschke i force pushed changes to my feature branch, but just noticed you had some kind of formatting changes that got squashed, and i can't restore. what were they?

File: moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp (hunk 4 above)
Comment (davetcoleman) [reply moveit_experimental/moveit_jog_arm/test/jog_cpp_interface_test.cpp]: i added sleep

**Comment by rhaschke** [comment]
> I force pushed changes to my feature branch

That's why one should always use [`--force-with-lease`](https://stackoverflow.com/questions/52823692/git-push-force-with-lease-vs-force) :wink:
I can recover. Will have a look asap. Thanks for the note.

**Comment by davetcoleman** [comment]
That's a cool trick I hadn't heard of!

**Pull request merged**